### PR TITLE
Sockets - Nara

### PR DESCRIPTION
**Pull request by bonara**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | Pushing elements to an array, conditional operators, creating JS objects. 
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | I searched for a JS method that is similar to each_with_index and was able to use a for in loop. I also found that JS had  a concat method to combine two arrays.  
What was a challenge you were able to overcome on this assignment? | Delete_at index was little bit tricky in JS. Also, since 'return' doesn't work in the forEach loop, I had to put the conditional statements outside of the loop. 
What has been interesting and positive about learning a new programming language? | Since I knew how things worked in Ruby, it was easier to search for similar methods in JavaScript. 
What is something to focus on your learnings in JavaScript in the next week? | JavaScript data types, its methods, syntax, and variable scopes.

### CODE REVIEW

File: src/adagrams.js [hunk 1]
```diff
@@ -1,8 +1,86 @@
...
+    const letterPool = { A: 9, B: 2, C: 2, D: 4, E: 12, F: 2, G: 3, H: 2, I: 9, 
+                        J: 1, K: 1, L: 4, M: 2, N: 6, O: 8, P: 2, Q: 1, R: 6, 
+                        S: 4, T: 6, U: 4, V: 2, W: 2, X: 1, Y: 2, Z: 1 };
+    const letterArray = [];
+    const drawn = [];
+    
+    for (let key in letterPool){
+      for( let i = 0; i < letterPool[key]; i += 1){
+        letterArray.push(key);
+      }
```
Comment (droberts-sea) [inline src/adagrams.js:12]: I really like that you store the letter frequencies and then build the pool on the fly - this is much easier to read than a giant array full of 9 'A's, 2 'B's, etc.

File: src/adagrams.js [hunk 2]
```diff
@@ -1,8 +1,86 @@
...
+    if (word.length >= 7){
+      points += 8;
+    }
+    return points;
   },
+
+  highestScoreFrom(words){
+    let winning = {};
+    let bestScore = 0;
+    let bestWord;
```
Comment (droberts-sea) [inline src/adagrams.js:64]: `winning` could be a constant here, since you never reassign it.

**Comment by droberts-sea** [comment]
# JS Adagrams
## What We're Looking For



Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | yes
Small commits with meaningful commit messages | yes
**Code Requirements** |
`drawLetters` method | yes
Uses appropriate data structure to store the letter distribution | yes
All tests for `drawLetters` pass | yes
`usesAvailableLetters` method | yes
All tests for `usesAvailableLetters` pass | yes
`scoreWord` method | yes
Uses appropriate data structure to store the letter scores | yes
All tests for `scoreWord` pass | yes
`highestScoreFrom` method | yes
Appropriately handles edge cases for tie-breaking logic | yes
All tests for `highestScoreFrom` pass | yes
**Overall** | Great job overall! This code is clean and well-organized, and it feels like the JavaScript unit is off to a solid start. Keep up the hard work!
